pyrit/memory/azure_sql_memory.py

```
@@ -0,0 +1,135 @@
+                # create token credential
+                azure_token = azure_credentials.get_token(self.TOKEN_URL)
+                raw_token = azure_token.token.encode("utf-16-le")
+                token_struct = struct.pack(f"<I{len(raw_token)}s", len(raw_token), raw_token)
+                # apply it to keyword arguments
+                cparams["attrs_before"] = {self.SQL_COPT_SS_ACCESS_TOKEN: token_struct}
+            return engine
+
+    def _create_tables_if_not_exist(self):
+        """
+        Creates all tables defined in the Base metadata, if they don't already exist in the database.
+
+        Raises:
+            Exception: If there's an issue creating the tables in the database.
+        """
+        try:
+            # Using the 'checkfirst=True' parameter to avoid attempting to recreate existing tables
+            Base.metadata.create_all(self.engine, checkfirst=True)
+        except Exception as e:
+            logger.error(f"Error during table creation: {e}")
+
+    def _add_embeddings_to_memory(self, *, embedding_data: list[EmbeddingData]) -> None:
+        return super()._add_embeddings_to_memory(embedding_data=embedding_data)
+
+    def _get_prompt_pieces_by_orchestrator(self, *, orchestrator_id: int) -> list[PromptRequestPiece]:
+        return super()._get_prompt_pieces_by_orchestrator(orchestrator_id=orchestrator_id)
+
+    def _get_prompt_pieces_with_conversation_id(self, *, conversation_id: str) -> list[PromptRequestPiece]:
+        return super()._get_prompt_pieces_with_conversation_id(conversation_id=conversation_id)
+
+    def add_request_pieces_to_memory(self, *, request_pieces: list[PromptRequestPiece]) -> None:
+        return super().add_request_pieces_to_memory(request_pieces=request_pieces)
+
+    def add_scores_to_memory(self, *, scores: list[Score]) -> None:
+        return super().add_scores_to_memory(scores=scores)
+
+    def dispose_engine(self):
+        return super().dispose_engine()
```

These functions are not yet implemented. Do you plan to move the concrete functionality from the DuckDB implementation to the memory interface and then leverage it in both DuckDB and Azure SQL modules? If that's the case, that's fantastic!

---

For the first PR, if not implemented here, can you throw an exception rather than calling super?